### ISSUE TITLE
Cache Invalidate Remote WASM Filter

### ISSUE THREAD

**Issue by NicholasLYang** [istio/istio#32743]
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**
When using [dynamically downloaded WASM modules](https://istio.io/latest/docs/ops/configuration/extensibility/wasm-module-distribution/), we want to have the ability to invalidate the cache when we update the code.

Specifically, we'd like to have a way to refresh the remote WASM filter and push that update down to the proxies.

**Describe alternatives you've considered**
Set a shorter purge interval, however that will affect performance.

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[x] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by NicholasLYang** [comment]
Sure. Can the checksum be updated without restarting the proxy? I'm probably misunderstanding here

**Comment by bianpengyuan** [comment]
Yeah you are only updating EnvoyFilter config, which is part of xds config pushed dynamically by control plane to the proxy.
